pkg/bisect/bisect.go

```diff
@@ -85,11 +87,26 @@ func Run(cfg *Config) ([]*vcs.Commit, *report.Report, error) {
 	if !ok {
 		return nil, nil, fmt.Errorf("bisection is not implemented for %v", cfg.Manager.TargetOS)
 	}
+	commitEnv, ok := repo.(vcs.EnvForCommitOverridable)
+	if !ok {
+		return nil, nil, fmt.Errorf("repo does not implement EnvForCommit")
+	}
 	env := &env{
-		cfg:      cfg,
-		repo:     repo,
-		bisecter: bisecter,
+		cfg:       cfg,
+		repo:      repo,
+		bisecter:  bisecter,
+		commitEnv: commitEnv,
+	}
+	if env.inst, err = instance.NewEnv(&cfg.Manager); err != nil {
+		return nil, nil, err
+	}
+	if env.head, err = env.repo.CheckoutBranch(cfg.Kernel.Repo, cfg.Kernel.Branch); err != nil {
+		return nil, nil, err
 	}
+	return RunImpl(cfg, env)
+}
+
+func RunImpl(cfg *Config, env *env) ([]*vcs.Commit, *report.Report, error) {
```

Comment:
This function should not be exported.

Reply:
env already contains cfg, so no need to accept it separately.
However, I think it's better to make env creation part of RunImpl as well, so that we test more code.
RunImpl could accept Config, Repo and Bisecter, and it seems to be enough to create env.